sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -190,6 +190,13 @@ private[hive] class HiveClientImpl(
     if (clientLoader.cachedHive != null) {
       Hive.set(clientLoader.cachedHive.asInstanceOf[Hive])
     }
+    // Hive 2.3 will set UDFClassLoader to hiveConf when initializing SessionState
+    // since HIVE-11878, and ADDJarCommand will add jars to clientLoader.classLoader.
+    // For this reason we cannot load the jars added by ADDJarCommand because of class loader
+    // got changed. We reset it to clientLoader.ClassLoader here.
+    if (HiveUtils.isHive23) {
+      state.getConf.setClassLoader(clientLoader.classLoader)
+    }
     SessionState.start(state)
```

Comment:
@AngersZhuuuu I think we need to add another test case simily to [`HiveSparkSubmitSuite#SPARK-8368: includes jars passed in through --jars`](https://github.com/apache/spark/blob/016e1b491c9099063df8e2e76bc58d64fc369490/sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSparkSubmitSuite.scala#L110-L127) but with `spark.sql.hive.metastore.version=3.1` and `spark.sql.hive.metastore.jars=maven`.

Reply:
Sorry. It's another bug: https://issues.apache.org/jira/browse/SPARK-29254

Reply:
SPARK-29254 is a different issue.